scripts/frontmatter-validator/frontmatter-validator.js

```diff
@@ -29,83 +29,113 @@ function initializeValidator() {
 function loadSchema(schemaPath) {
   const args = process.argv.slice(2);
   let fullPath = `scripts/frontmatter-validator/${schemaPath}`;
-  fullPath = args.includes("--file")
+  fullPath = args.includes('--file')
     ? `scripts/frontmatter-validator/${schemaPath}`
     : schemaPath;
 
   // todo fix for non file input
-  const json = JSON.parse(fs.readFileSync(fullPath, "utf8"));
+  const json = JSON.parse(fs.readFileSync(fullPath, 'utf8'));
   return json;
 }
 
-function matchSchema(filePath) {
-  const isRule = filePath.endsWith("rule.md");
+function determineCategory(filePath) {
+  const isRule = filePath.endsWith('rule.md');
   const isInCategories =
-    filePath.includes("/categories") &&
-    !filePath.endsWith("/categories/index.md");
-  const isIndexFile = filePath.endsWith("index.md");
-
+    filePath.includes('/categories') &&
+    !filePath.endsWith('/categories/index.md');
+  const isIndexFile = filePath.endsWith('index.md');
   if (isRule) {
-    return validator.getSchema("rule");
+    return 'rule';
   }
-
   if (isInCategories) {
     if (!isIndexFile) {
-      return validator.getSchema("category");
+      return 'category';
     } else {
-      return validator.getSchema("top_category");
+      return 'top_category';
     }
   }
 }
 
+function getMissingSpaceErrors(frontmatterContents, schema) {
+  const errors = [];
+  if (!frontmatterContents) return [];
+  for (const field of Object.keys(schema.properties)) {
+    const regex = new RegExp(`[\\s^](${field}:)\\S`);
+    const match = frontmatterContents.match(regex);
+    if (frontmatterContents.match(regex)) {
+      errors.push(`space missing for field: '${match[1]}'`);
```

Comment:
Can we make this a bit clearer? Maybe something like, 

---

A space is missing after the `:` in the `title` field. Please add a space between `:` and the `title` value.

Reply:
See comment

https://github.com/SSWConsulting/SSW.Rules.Content/pull/9531#discussion_r1827218467